DatadogLogs/Sources/Logs.swift

```diff
@@ -0,0 +1,110 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// iOS Log Collection
+///
+/// Send logs to Datadog from your iOS applications with Datadog’s dd-sdk-ios client-side logging library and leverage the following features:
+/// - Log to Datadog in JSON format natively.
+/// - Use default and add custom attributes to each log sent.
+/// - Record real client IP addresses and User-Agents.
+/// - Leverage optimized network usage with automatic bulk posts.
+public struct Logs {
+    /// The Logs general configuration.
+    ///
+    /// This configuration will be applied to all Logger instances.
+    public struct Configuration {
+        public typealias EventMapperClosure = (LogEvent) -> LogEvent?
+        /// Sets the sampling rate for logging.
+        ///
+        /// The sampling rate must be a value between `0` and `100`. A value of `0` means no logs will be processed, `100`
+        /// means all logs will be processed.
+        ///
+        /// By default sampling is disabled, meaning that all logs are being processed).
+        public var sampleRate: Float
+
+        /// Overrides the custom server endpoint where Logs are sent.
+        public var customEndpoint: URL?
+
+        /// Overrides the main bundle instance.
+        public var bundle: Bundle = .main
+
+        /// Overrides the current process info.
+        internal var processInfo: ProcessInfo = .processInfo
+
+        /// Sets the custom mapper for `LogEvent`. This can be used to modify logs before they are send to Datadog.
+        ///
+        /// - Parameter mapper: the closure taking `LogEvent` as input and expecting `LogEvent` as output.
+        /// The implementation should obtain a mutable version of the `LogEvent`, modify it and return it. Returning `nil` will result
+        /// with dropping the Log event entirely, so it won't be send to Datadog.
+        public mutating func eventMapper(_ mapper: @escaping EventMapperClosure) {
+            self.mapper = SyncLogEventMapper(mapper)
+        }
```

Comment:
blocker (simplicity)/ Do we need to make it that complex 🙂? Simply:
```swift
public var eventMapper: LogEventMapper?
```
can be obvious enough. It will offer both **setter** and **getter** and will be consistent with closure-based APIs in other products (_keep it simple_ is a good baseline here).

If Flutter SDK needs different mapper, we can resolve it when used:
```swift
let feature = LogsFeature(
   logEventMapper: config._internalEventMapper ?? config.eventMapper
}
```

Reply:
Indeed, that's a good point 👍